clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/Transport.java

```diff
@@ -3,9 +3,13 @@
 
 package com.microsoft.aspnet.signalr;
 
+import java.util.concurrent.CompletableFuture;
+
 public interface Transport {
     void start() throws Exception;
+    CompletableFuture startAsync() throws Exception;
     void send(String message) throws Exception;
+    CompletableFuture sendAsync(String message) throws Exception;
     void setOnReceive(OnReceiveCallBack callback);
     void onReceive(String message) throws Exception;
     void stop();
```

Comment:
The interface should be async here, even if the implementation is fully sync and just returns a completed future.

Reply:
`WebSocketClient.closeConnection` can probably block though so it should be wrapped in `CompletableFuture.runAsync` anyway.